specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json

```diff
@@ -0,0 +1,358 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-06-01-preview",
+    "title": "SqlManagementClient",
+    "description": "The Azure SQL Database management API provides a RESTful set of web APIs that interact with Azure SQL Database services to manage your databases. The API enables users to create, retrieve, update, and delete databases, servers, and other entities."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/managedInstances/{managedInstanceName}/operations/{operationId}/cancel": {
```

Comment:
You need a GET API for individual operations to be compliant with the RPC especially if you are now allowing people to cancel individual operations.  A comparison is operations on a template deployment:  https://docs.microsoft.com/en-us/rest/api/resources/deploymentoperations/get

Reply:
This API for cancel was done in a manner to be compliant with SQL DB and they don't have an API fro getting one particular operation. 
Also can you please explain if there is a requirement to create a new API or if we can add not-required parameter in existing API for listing all operations-operationId which will allow customer to get one particular operation.

Reply:
Existing APIs not following [the RPC ](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#nested-resources) shouldn't be a reason to continue that trend.  Services within Azure rely on RPs following the RPC so they can work with resources in a generic way without needing to know the quirks of any specific API.

I have not seen an example in existing swagger specs where someone modeled a LIST and GET operation in the same operation via an optional name parameter.  The response model would be different so I don't imagine that is a good idea.

Reply:
I don't think that we understood each other entirely. What I was thinking of doing is adding new key in existing API for getting all operations.
Current call of API:
https://management.azure.com/subscriptions/a8c9a924-06c0-4bde-9788-e7b1370969e1/resourceGroups/v-urmila/providers/Microsoft.Sql/managedInstances/v-urmila-mi-test/operations?api-version=2018-06-01-preview
which returns all operations to customer.
And adding a key _operationId_ the same API can be called like this:
https://management.azure.com/subscriptions/a8c9a924-06c0-4bde-9788-e7b1370969e1/resourceGroups/v-urmila/providers/Microsoft.Sql/managedInstances/v-urmila-mi-test/operations/a64077e8-43e6-47b6-a6ef-9c2f40fac38b?api-version=2018-06-01-preview